lightning/src/ln/outbound_payment.rs

```diff
@@ -271,6 +378,25 @@ impl OutboundPayments {
 		}
 	}
 
+	pub(super) fn send_payment<R: Deref, ES: Deref, NS: Deref, F>(
+		&self, payment_hash: PaymentHash, payment_secret: &Option<PaymentSecret>, payment_id: PaymentId,
+		retry_strategy: Retry, route_params: RouteParameters, router: &R,
+		first_hops: Vec<ChannelDetails>, inflight_htlcs: InFlightHtlcs, entropy_source: &ES,
+		node_signer: &NS, best_block_height: u32, send_payment_along_path: F
+	) -> Result<(), PaymentSendFailure>
+	where
+		R::Target: Router,
+		ES::Target: EntropySource,
+		NS::Target: NodeSigner,
```

Comment:
No need to use references to a `Deref`. Just use `Deref` and update the call sites to do `&*` if needed. Then the tests don't need `&&`.

Likewise elsewhere.